README.md

```diff
@@ -102,6 +102,22 @@ Note that for the deployment to the S3 bucket to succeed, you must make the foll
 
 You should see the deployed app at http://BUCKET-NAME.s3-website-REGION.amazonaws.com such as http://mybucket.s3-website-us-east-1.amazonaws.com
 
+### Payara Deployment
+
+This option will build and deploy the application to a remote Payara server.
+
+This option is intended for an "all-in-one" solution, where the Dataverse backend application and the frontend application run on the same Payara server.
+
+For this workflow to work, a GitHub environment must be configured with the following environment secrets:
+
+- PAYARA_INSTANCE_HOST
+- PAYARA_INSTANCE_USERNAME
+- PAYARA_INSTANCE_SSH_PRIVATE_KEY
+
+It is important that the remote instance is correctly pre-configured, with the Payara server running, and a service account for Dataverse with the corresponding SSH key pair established.
```

Comment:
I'm happy to test this to see if this works but I'm not in the habit of allowing the user that Payara runs as (`dataverse`, usually) ssh into the server.

It strikes me as a bit less secure than what I think is out typical setup where the service user can't ssh into the server.

My usual pattern is:

- ssh rocky@host
- sudo -i
- su - dataverse

Once I'm the `dataverse` users I deploy to Payara.

Reply:
The service account has limited permissions within the server, and is only intended for Dataverse applications management. We create a key pair for that service account to be able to access the instance and perform ssh remote operations on the Payara server. 

I personally see the service account ssh approach as more secure, since we don't use a general purpose user to access the instance.

Let me know what you think. We can run a quick zoom if you want.

Thanks!

Reply:
No it's fine. After discussing it a bit internally with @poikilotherm and @donsizemore on Slack I'm fine with this since the use case is testing/QA.

I'm taking a few notes. On the server I spun up with dataverse-ansible the `id_rsa` file is empty, so I'm overwriting it and then adding it to `authorized_keys`.

```
[dataverse@ip-172-31-5-10 ~]$ file .ssh/id_rsa
.ssh/id_rsa: empty
[dataverse@ip-172-31-5-10 ~]$ ssh-keygen 
Generating public/private rsa key pair.
Enter file in which to save the key (/home/dataverse/.ssh/id_rsa): 
/home/dataverse/.ssh/id_rsa already exists.
Overwrite (y/n)? y
Enter passphrase (empty for no passphrase): 
Enter same passphrase again: 
Your identification has been saved in /home/dataverse/.ssh/id_rsa.
Your public key has been saved in /home/dataverse/.ssh/id_rsa.pub.
The key fingerprint is:
SHA256:ZeQKzhqurLX70NN/qlx4gZ724eTle82yKhgoRKvWy8w dataverse@ip-172-31-5-10.ec2.internal
The key's randomart image is:
+---[RSA 3072]----+
|          .      |
|  .      o       |
| . .  .   +      |
|  o  o o +       |
| o. ..+ S        |
|...+.=.o .       |
|. *.* *o= .  o   |
| o E +.O.+. o o  |
|..=o. o.*+++.o   |
+----[SHA256]-----+
[dataverse@ip-172-31-5-10 ~]$ 
[dataverse@ip-172-31-5-10 ~]$ cat .ssh/id_rsa.pub >> ~/.ssh/authorized_keys 
[dataverse@ip-172-31-5-10 ~]$ 
```

Now from my Mac I'm able to ssh into the server as the `dataverse` user.

```
ssh -i dev1-dataverse.pem dataverse@dev1.dataverse.org
```